// Generated by CoffeeScript 1.12.7
(function() {
  var assert;

  assert = require('assert');

  Function.prototype.assert_Throws = function(callback) {
    var expected_Message, message, onError;
    message = "[assert_Throws]";
    if (typeof callback === 'string') {
      expected_Message = callback;
      callback = function(error) {
        var error_Message;
        error_Message = error.message || error;
        return error_Message.assert_Is(expected_Message, "Expected exception error to be '" + expected_Message + "' but it was '" + error_Message + "'");
      };
    }
    onError = (function(_this) {
      return function(error) {
        if (callback) {
          callback(error);
        }
        return true;
      };
    })(this);
    assert.throws(this, onError, message);
    return this;
  };

  Function.prototype.assert_Not_Throws = function() {
    var message;
    message = "[assert_Not_Throws]";
    assert.doesNotThrow(this, message);
    return this;
  };

  Function.prototype.assert_Is_Function = function() {
    var message, test;
    message = "[assert_Is_Function]";
    test = typeof this;
    assert.equal(test, 'function', message);
    return this;
  };

}).call(this);
